src/rules/selector-no-redundant-parent/index.js

```diff
@@ -0,0 +1,27 @@
+import { utils } from "stylelint"
+
+export const ruleName = "selector-no-redundant-parent"
+
+export const messages = utils.ruleMessages(ruleName, {
+  rejected: "Unnecessary parent selector (&)",
+})
+
+export default function (actual) {
+  return function (root, result) {
+    const validOptions = utils.validateOptions(result, ruleName, { actual })
+    if (!validOptions) { return }
+
+    root.walkRules(rule => {
+      const unnecessaryAmpersandRegex = /^&(\s+)?>?(\s+)?(\w+|\s\.)/
+
+      if (rule.selector.trim() === "&" || unnecessaryAmpersandRegex.test(rule.selector.trim())) {
```

Comment:
Maybe you'd consider it too much of an edge case, but what if there's an `&` in an attribute, e.g. `div[id="foo&bar"]`?

Also: have you looked into postcss-selector-parser? Maybe not necessary if your regexp is working great, but might turn out more accurate in the end if you start running into edge cases. Just an idea.